win32ss/user/user32/controls/listbox.c

```diff
@@ -2560,18 +2654,15 @@ static BOOL LISTBOX_Create( HWND hwnd, LPHEADCOMBO lphc )
     if (descr->style & LBS_EXTENDEDSEL) descr->style |= LBS_MULTIPLESEL;
     if (descr->style & LBS_MULTICOLUMN) descr->style &= ~LBS_OWNERDRAWVARIABLE;
     if (descr->style & LBS_OWNERDRAWVARIABLE) descr->style |= LBS_NOINTEGRALHEIGHT;
+    if ((descr->style & (LBS_OWNERDRAWFIXED | LBS_HASSTRINGS | LBS_SORT)) != LBS_OWNERDRAWFIXED)
+        descr->style &= ~LBS_NODATA;
 
-    //// ReactOS
-    /* A no-data list box must also have the LBS_OWNERDRAWFIXED style, but must
```

Comment:
Is this thing now correctly handled by the new code above (lines 2657-2658) and below (line 2664) ?

Reply:
Yeah the logic is ordered a little differently but it evaluates to the same idea, 

if im wrong about this someone let me know!
But this is "the same picture"